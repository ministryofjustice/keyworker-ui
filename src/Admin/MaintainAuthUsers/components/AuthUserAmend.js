import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'
import InputField from '@govuk-react/input-field'
import Button from '@govuk-react/button'
import GridRow from '@govuk-react/grid-row'
import GridCol from '@govuk-react/grid-col'
import { BLACK, GREY_3 } from 'govuk-colours'
import { SPACING } from '@govuk-react/constants'
import { errorType } from '../../../types'
import { lookupMeta } from '../../../govuk-helpers'

export const ButtonContainer = styled('div')`
  display: flex;
  flex-direction: row;
  button {
    margin-right: ${SPACING.SCALE_2};
  }
`

const AuthUserAmend = ({ handleAmend, handleChange, handleCancel, error, email }) => (
  <GridRow>
    <GridCol setWidth="two-thirds">
      <form onSubmit={handleAmend} noValidate>
        <InputField
          meta={lookupMeta('email', error)}
          htmlFor="email"
          mb={6}
          input={{
            id: 'email',
            name: 'email',
            onChange: handleChange,
            autoComplete: 'alter-email',
            type: 'email',
            spellCheck: false,
            defaultValue: email,
          }}
        >
          Email address
        </InputField>
        <ButtonContainer>
          <Button type="submit" data-qa="amend-button">
            Amend
          </Button>
          <Button
            type="button"
            data-qa="cancel-button"
            onClick={handleCancel}
            buttonColour={GREY_3}
            buttonTextColour={BLACK}
          >
            Cancel
          </Button>
        </ButtonContainer>
      </form>
    </GridCol>
  </GridRow>
)

AuthUserAmend.propTypes = {
  email: PropTypes.string.isRequired,
  handleAmend: PropTypes.func.isRequired,
  handleChange: PropTypes.func.isRequired,
  handleCancel: PropTypes.func.isRequired,
  error: errorType,
}

AuthUserAmend.defaultProps = {
  error: [],
}

export default AuthUserAmend
