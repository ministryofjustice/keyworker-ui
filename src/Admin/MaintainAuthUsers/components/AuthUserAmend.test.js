import React from 'react'
import renderer from 'react-test-renderer'
import AuthUserAmend from './AuthUserAmend'

describe('Auth Amend form', () => {
  const stubFunc = () => {}
  describe('Form', () => {
    it('should render correctly', () => {
      const wrapper = renderer
        .create(
          <AuthUserAmend handleChange={stubFunc} handleAmend={stubFunc} handleCancel={stubFunc} email="john.smith" />
        )
        .toJSON()
      expect(wrapper).toMatchSnapshot()
    })
  })
})
