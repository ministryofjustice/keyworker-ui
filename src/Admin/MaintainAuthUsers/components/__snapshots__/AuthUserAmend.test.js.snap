// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Auth Amend form Form should render correctly 1`] = `
<div
  className="sc-Axmtr hjHhIB"
>
  <div
    className="sc-AxmLO czuJjP"
  >
    <form
      noValidate={true}
      onSubmit={[Function]}
    >
      <label
        className="sc-AxjAm bnkTnU"
        htmlFor="email"
      >
        <span
          className="sc-AxirZ lbZqdG"
        >
          Email address
        </span>
        <input
          autoComplete="alter-email"
          className="sc-AxhUy hUbURi"
          defaultValue="john.smith"
          id="email"
          name="email"
          onChange={[Function]}
          spellCheck={false}
          type="email"
        />
      </label>
      <div
        className="sc-fzozJi lkMuvo"
      >
        <button
          className="sc-AxgMl dzrPxP"
          data-qa="amend-button"
          disabled={false}
          type="submit"
        >
          Amend
        </button>
        <button
          className="sc-AxgMl iErfzZ"
          data-qa="cancel-button"
          disabled={false}
          onClick={[Function]}
          type="button"
        >
          Cancel
        </button>
      </div>
    </form>
  </div>
</div>
`;
